pkg/config/engine/containerd/containerd.go

```diff
@@ -72,45 +88,50 @@ func New(opts ...Option) (engine.Interface, error) {
 		return nil, fmt.Errorf("failed to load config: %v", err)
 	}
 
-	cfg := &Config{
-		Tree:                  tomlConfig,
-		Logger:                b.logger,
-		RuntimeType:           b.runtimeType,
-		UseDefaultRuntimeName: b.useLegacyConfig,
-		ContainerAnnotations:  b.containerAnnotations,
+	configVersion, err := b.parseVersion(tomlConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse config version: %w", err)
 	}
 
-	version, err := cfg.parseVersion(b.useLegacyConfig)
-	if err != nil {
-		return nil, fmt.Errorf("failed to parse config version: %v", err)
+	cfg := &Config{
+		Tree:                 tomlConfig,
+		Version:              configVersion,
+		Logger:               b.logger,
+		RuntimeType:          b.runtimeType,
+		UseLegacyConfig:      b.useLegacyConfig,
+		ContainerAnnotations: b.containerAnnotations,
 	}
-	switch version {
+
+	switch configVersion {
 	case 1:
 		return (*ConfigV1)(cfg), nil
 	case 2:
+		cfg.CRIRuntimePluginName = "io.containerd.grpc.v1.cri"
```

Comment:
The intent of this switch statement is just to cast the config to the correct type, so instead of doing this here, should we set some local based on the version above and set `CRIRuntimePluginName` in the instantiation directly?

I think we could add `b.criRuntimePluginName(version)` or something to make this cleaner.